x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/RBACEngineTests.java

```diff
@@ -1485,24 +1481,27 @@ public void testBackingIndicesAreIncludedForAuthorizedDataStreams() {
         }
 
         SearchRequest request = new SearchRequest("*");
-        Set<String> authorizedIndices = RBACEngine.resolveAuthorizedIndicesFromRole(
+        AuthorizedIndices authorizedIndices = RBACEngine.resolveAuthorizedIndicesFromRole(
             role,
             getRequestInfo(request, SearchAction.NAME),
-            lookup
+            lookup,
+            () -> ignore -> {}
         );
-        // The authorized indices is the lazily loading set implementation
-        assertThat(authorizedIndices, instanceOf(RBACEngine.AuthorizedIndicesSet.class));
-        assertThat(authorizedIndices, hasItem(dataStreamName));
+        assertThat(authorizedIndices, instanceOf(AuthorizedIndices.class));
```

Comment:
Nit: This assertion is not necessary anymore since the AuthorizedIndices is no longer a Set implementation and the type is guaranteed statically.

Reply:
Pushed [531245e](https://github.com/elastic/elasticsearch/pull/91618/commits/531245e18824215ec0f6f316a0bfc508af2bd895)